splinterd/src/main.rs

```diff
@@ -485,25 +467,27 @@ fn main() {
 
     let matches = app.get_matches();
 
-    let log_level = match matches.occurrences_of("verbose") {
+    let clap_log_level = match matches.occurrences_of("verbose") {
         0 => log::LevelFilter::Warn,
         1 => log::LevelFilter::Info,
         2 => log::LevelFilter::Debug,
         _ => log::LevelFilter::Trace,
     };
 
-    let mut log_spec_builder = LogSpecBuilder::new();
-    log_spec_builder.default(log_level);
-    log_spec_builder.module("hyper", log::LevelFilter::Warn);
-    log_spec_builder.module("tokio", log::LevelFilter::Warn);
+    let encoder = PatternEncoder::new("[{d(%Y-%m-%d %H:%M:%S%.3f)}] T[{T}] {l} [{M}] {m}\n");
+    let stdout = log4rs::append::console::ConsoleAppender::builder()
+        .encoder(Box::new(encoder))
+        .build();
+    let config = log4rs::Config::builder()
+        .appender(Appender::builder().build("stdout", Box::new(stdout)))
+        .logger(Logger::builder().build("hyper", log::LevelFilter::Warn))
+        .logger(Logger::builder().build("tokio", log::LevelFilter::Warn));
     #[cfg(feature = "https-bind")]
-    log_spec_builder.module("h2", log::LevelFilter::Warn);
-
-    Logger::with(log_spec_builder.build())
-        .format(log_format)
-        .log_target(flexi_logger::LogTarget::StdOut)
-        .start()
-        .expect("Failed to create logger");
+    let config = config.logger(Logger::builder().build("h2", log::LevelFilter::Warn));
+    let log_config = config.build(Root::builder().appender("stdout").build(clap_log_level));
+    if let Ok(lc) = log_config {
```

Comment:
What should happen in the error case?

Reply:
If this commit actually could have an error case we would probably want it to either panic or print a message and continue.  As written it shouldn't ever throw any errors because its not a dynamic configuration. There's basically one compile time option and 4 well established and tested runtime options. I can add a panic/expect if you'd prefer that instead though.

Reply:
Perhaps an else block with an `unreachable!()` call and a comment explaining why

Reply:
That seems reasonable, Ill add push changes in a minute